v/vm.c

```diff
@@ -296,5 +364,13 @@ void vm_boot(uintptr_t test_addr)
   trapframe_t tf;
   memset(&tf, 0, sizeof(tf));
   tf.epc = test_addr - DRAM_BASE;
+  
+  //user_llpt[VPN0(tf.epc)] = ((pte_t)test_addr/RISCV_PGSIZE << PTE_PPN_SHIFT) | PTE_V | PTE_R | PTE_W | PTE_X | PTE_A | PTE_D;
+  for (long i = 0; i < usize; i++)
```

Comment:
Before these changes, the user pages were not actually in the same location as the physical pages in the mem. 
And on a page-fault, it copied those physical pages to the allocated user addresses.
I am not sure if one approach is better than the other, but in the old one, user mappings can be random and independent of how the code is loaded in the mem.

Reply:
Certainly original way will allow us to test page fault.